src/funding/funding.ts

```diff
@@ -160,9 +164,18 @@ export class Funding {
     // show a modal confirming the user wants to withdraw their funds
     const dialogResult = await this.alertService.showAlert(withdrawModal);
     if (!dialogResult.wasCancelled) {
-      //TODO wire up the withdraw method
-      this.eventAggregator.publish("handleException", new EventConfigException("This method is not implemented", EventMessageType.Exception));
+      //withdraw the tokens
+      const withdrawTransaction = await this.deal.withdrawTokens(transaction.dao, transaction.depositId);
+      if (withdrawTransaction){
+        //tokens were withdrawn successfully
+        this.eventAggregator.publish("handleInfo", new EventConfig("Your deposit has been withdrawn", EventMessageType.Info, "Withdrawn"));
+      } else {
+        //an error occurred while trying to withdraw tokens
+        this.eventAggregator.publish("handleException", new EventConfigException("An error occurred while trying to withdraw. Please try again.", EventMessageType.Exception));
```

Comment:
Do "handleFailure" with just a string as the message.

Reply:
Done